vue.config.js

```diff
@@ -0,0 +1,66 @@
+const path = require('path')
+
+/**
+ * @type {import('@vue/cli-service').ProjectOptions}
+ */
+module.exports = {
+  assetsDir: 'static',
+  runtimeCompiler: true,
```

Comment:
Do we need the runtime compiler? At least for my tests, the application seems to build and run fine without it.

Reply:
It depends how you define your Vue components. I'm almost sure that the demo app doesn't need it as you said.
It was added to reduce the risk to break some apps as the documentation states that [it only incurs a supplementary 10kb payload for the entire app](https://cli.vuejs.org/config/#runtimecompiler).
I don't remember exactly but I'm wondering if some unit tests were not dependent on it with some mount commands redefining the template of components. I'm not 100% sure though...

Reply:
So far I had no problem here with unit tests or the custom app I tested. I think the default templates turn it off, because it imposes a runtime overhead. Could you figure out, whether a runtime compile is supported by the current Wegue toolchain?

Reply:
Concerning this point, I made some more researches. It's true that Vue CLI since V3 turns this value as [false by default](https://github.com/vuejs/vue-cli/blob/v3.12.1/packages/@vue/cli-service/lib/options.js#L98-L99).
However, as stated above, this was set to reduce the risk to break some apps as the compiler IS currently bundled [in the latest Wegue version](https://github.com/wegue-oss/wegue/blob/6dd5206fd1858edda4615b306da34a0c2016f9bc/build/webpack.base.conf.js#L25).
This was also the default value [in the template of Vue CLI V2](https://github.com/vuejs-templates/webpack/blob/c29ab6a4679f74d970e025bb60c4602542eaff08/template/build/webpack.base.conf.js#L38).
So to answer your question, yes, current Wegue toolchain supports compilation at runtime and could potentially use it.

On the other side, I agree with you 100%, this shouldn't be used in default Wegue app and would only be used in very peculiar situations. This compiler is in fact automatically called when your code contains components which were not written in .vue files and were not processed by webpack beforehand. This can be useful if someone uses a component with dynamic templates for example.
I came on this [SO discussion](https://stackoverflow.com/questions/66393740/what-exactly-is-vues-runtime-only-build-and-how-does-it-differ-from-compiler-bu) where all I could write here is already stated very clearly.

To summarize, I would, as you suggested, have removed it personally but didn't want to include a potential breaking change. Perhaps removing it but mentioning it later in a migration note or something could be a good idea... just tell me what you want me to do and if I should do it now or in the next transition to V4...